### ISSUE TITLE
Certain preconditions broken in >= 3.39.x

### ISSUE THREAD

**Issue by wburningham** [go-task/task#1836]
## Environment info:
- Task version: `3.39.0`
- Operating system: `OS: macOS 14.6.1 23G93 arm64`
- Experiments enabled: `TASK_X_REMOTE_TASKFILES`

## Description:

Certain preconditions start failing in >= `3.39.0` that used to work in `3.38.0`.

## Steps to reproduce:

```
❯ cat taskfile.dist.yaml
---
version: 3

tasks:
  default:
    preconditions:
      - docker --version
    cmd: docker --version⏎                                                                                                         

❯ docker --version
Docker version 25.0.3, build 4debf41
```

## Expected results

The task should execute like it does in `v3.38.0`

```
❯ go run github.com/go-task/task/v3/cmd/task@v3.38.0
task: [default] docker --version
Docker version 25.0.3, build 4debf41
```

## Actual results

The task fails in a precondition in version >= `v3.39.0`

```
❯ go run github.com/go-task/task/v3/cmd/task@v3.39.2
task: `docker --version` failed
task: precondition not met
exit status 1

❯ go run github.com/go-task/task/v3/cmd/task@v3.39.0
task: `docker --version` failed
task: precondition not met
exit status 1
```

**Comment by vmaerten** [comment]
I cannot reproduced.
I tried with my linux but also with a Macbook M2. Can you reproduce in a CI (Github action ? Gitlab CI ?)

**Comment by wburningham** [comment]
Running `go run github.com/go-task/task/v3/cmd/task@v3.40.0` works as expected so it seems like it has been fixed with the recent 3.40 release. This can be closed.
